CHANGELOG.md

```diff
@@ -25,13 +34,35 @@ All notable changes to this project will be documented in this file, in reverse
 - [#37](https://github.com/zendframework/zend-di/pull/37) removes the use of `count()` 
   in `Zend\Di\CodeGenerator\FactoryGenerator::buildParametersCode()` to improve performance   
 
+- [#38](https://github.com/zendframework/zend-di/pull/38) adds strictness to
+  `Zend\Di\CodeGenerator\FactoryGenerator::generate()`:
+  - Adds `string` return type.
+  - Adds throw of `RuntimeException` on failures.  
+  
+- [#38](https://github.com/zendframework/zend-di/pull/38) removes inheritance of 
+  `Zend\Di\Resolver\AbstractInjection`:
+   - from `Zend\Di\Resolver\ValueInjection`
```

Comment:
Came across this, when double checking if there is a need for updating the migration guide as suggested by @froschdesign 

Actually this is a BC break:
- When users rely on and hint for `AbstractInjection`
- When using `get/setParameterName()`

One solution would be to make `AbstractInjection` implement `InjectionInterface` and extend `Value/TypeInjection` from the abstract class again. But this causes the following issues:
- The file level deprecation of the abstract class will break a lot of tests 
- Still a BC-Break: `InjectionInterface` introduces `toValue()`

To let only `Value/TypeInjection` implement `InjectionInterface` and extend from `AbstractInjection` will cause another BC-Break:
- Custom implementations that extend `AbstractInjection` will no longer work, since they do not implement the required interface.

I currently have no idea how to solve this without breaking BC on one way or the other. I'd tend to leave it like it is and document it, I doubt that anybody will be affected by this.

@weierophinney, @Ocramius: Any ideas/suggestions?

Reply:
The question to ask is: is there any situation in which a user would be type-hinting and/or providing an alternative implementation?

If the answer is no, it's not a BC break.

If the answer is yes, then you have another question: will leaving the inheritance as-is raise more problems for end-users than it will solve? If this is the case, it's a _justified_ BC break: the "happy path" should not raise errors under normal usage UNLESS there's an easy alternative for users to migrate to.

From what I can see, these various injection classes are _internal_, so changing the inheritance should not be a problem.